sw/device/lib/drivers/gpio/gpio.h

```diff
@@ -1,282 +1,369 @@
- * @return The result of the operation.
- */
-gpio_result_t gpio_init(gpio_params_t params, gpio_t *gpio);
-
-/**
- * Resets a GPIO device.
- *
- * Resets the given GPIO device by setting its configuration registers to
- * reset values. Disables interrupts, output, and input filter.
- *
- * @param gpio A GPIO handle.
- * @return The result of the operation.
- */
-gpio_result_t gpio_reset(const gpio_t *gpio);
-
-/**
- * Returns whether a particular pin's interrupt is currently pending.
- *
- * @param gpio A GPIO handle.
- * @param pin A GPIO pin.
- * @param[out] is_pending Out-param for whether the interrupt is pending.
- * @return The result of the operation.
- */
-gpio_result_t gpio_irq_is_pending(const gpio_t *gpio,
-                                          gpio_pin_t pin, bool *is_pending);
-
-/**
- * Returns a GPIO state representing which pins have interrupts enabled.
- *
- * @param gpio A GPIO handle.
- * @param[out] is_pending Out-param for which interrupts are pending.
- * @return The result of the operation.
- */
-gpio_result_t gpio_irq_is_pending_all(const gpio_t *gpio,
-                                              gpio_state_t *is_pending);
-
-/**
- * Acknowledges a particular pin's interrupt, indicating to the hardware that it
- * has
- * been successfully serviced.
- *
- * @param gpio A GPIO handle.
- * @param pin A GPIO pin.
- * @return The result of the operation.
- */
-gpio_result_t gpio_irq_acknowledge(const gpio_t *gpio,
-                                           gpio_pin_t pin);
-
-/**
- * Configures interrupt triggers for a set of pins.
- *
- * This function configures interrupt triggers, i.e. rising-edge, falling-edge,
- * level-high, and level-low, for the pins given by the mask. Note that
- * interrupt of the pin must also be enabled to generate interrupts.
- *
- * @param gpio A GPIO handle.
- * @param mask Mask that identifies the pins whose interrupt triggers will be
- * configured.
- * @param trigger New configuration of interrupt triggers.
- * @return The result of the operation.
+ * The different interrupt types can be set by user
+ */
+typedef enum gpio_intr_type
+{
+    GpioIntrEdgeRising,         /*!< Trigger on rising edge. */
+    GpioIntrEdgeFalling,        /*!< Trigger on falling edge. */
+    GpioIntrLevelLow,           /*!< Trigger when input is low. */
+    GpioIntrLevelHigh,          /*!< rigger when input is high. */
+    GpioIntrEdgeRisingFalling,  /*!< Trigger on rising and falling edges. */
+    GpioIntrEdgeRisingLevelLow, /*!< Trigger on rising edge or when the 
+    input is low. */
+    GpioIntrEdgeFallingLevelHigh,/*!< Trigger on falling edge or when the 
+    input is high. */
+} gpio_intr_type_t;
+
+/**
+ * The structure should be used by user for configuring gpio.
+ * it includes pin number, mode, enable sampling, enable intr, and its mode.
+ */
+typedef struct gpio_cfg
+{
+    gpio_pin_number_t pin;      /*!< pin number. */
+    gpio_pin_number_t mode;     /*!< pin mode. */
```

Comment:
can we define an enum with the different modes?

Reply:
It is already defined, I used gpio_pin_number_t than gpio_mode_t by mistake for second field